### PR TITLE
update modularization slides

### PR DESCRIPTION
**Pull request by DaniBodor**
I'm still not completely happy with the stateful/stateless function example. There are pros and cons to both:

The main problem I have with the old example is that I don't think it's the kind of mistake that many people would make (explicitly declaring a global variable). I think that this will result in people not remembering/realizing that this can be an issue (in their case) as well.
What is good about that example is that follows suit with the temperature conversion theme, and that it truly changes the outside environment in a way that can lead to problems down the line, if one assumes that that parameter hasn't changed due to it being used in the function.

The advantage of the new example is that it is a typical mistake that many people can make and will be more recognizable. However, the environment that it changes is kind of a "hidden environment" that is only used when the function is called in subsequent iterations.

### CODE REVIEW

**Comment by lyashevska** [comment]
This is a part of this: https://github.com/esciencecenter-digital-skills/good-practices-in-research-software-development/issues/27

**Comment by DaniBodor** [comment]
> I'm still not completely happy with the stateful/stateless function example. There are pros and cons to both:
> 
> The main problem I have with the old example is that I don't think it's the kind of mistake that many people would make (explicitly declaring a global variable). I think that this will result in people not remembering/realizing that this can be an issue (in their case) as well. What is good about that example is that follows suit with the temperature conversion theme, and that it truly changes the outside environment in a way that can lead to problems down the line, if one assumes that that parameter hasn't changed due to it being used in the function.
> 
> The advantage of the new example is that it is a typical mistake that many people can make and will be more recognizable. However, the environment that it changes is kind of a "hidden environment" that is only used when the function is called in subsequent iterations.

I think we should stick to talking about pure/impure functions rather than stateless/stateful. These are separate (albeit related) concepts. I don't think we should be saying that people should never use stateful functions, as this is a design choice, but I do think we can say that it's bad practice to use impure functions.

In d16c7a0 changed the example to a kind of mix of the previous example and the one I made in abd7237

File: code_modularization/00_title.md [hunk 1]
```diff
@@ -239,43 +310,42 @@ def fahrenheit_to_celsius(temp_f):
...
-        if degrees < -273:
-            if degrees < -459:
-                print("This temperature is impossible.")
-            else:
-                print("This temperature is likely Fahrenheit.")
-        else:
-            print("This temperature is either Celsius or Fahrenheit.")
-    else:
-        print("This temperature is in Kelvin, Celsius, or Fahrhenheit.")
+def sum_even_numbers(bottom, top):
```
Comment (lyashevska) [inline code_modularization/00_title.md:313]: Good examples, but why not just keeping checkTemperature()?

File: code_modularization/00_title.md [hunk 2]
```diff
@@ -239,43 +310,42 @@ def fahrenheit_to_celsius(temp_f):
...
-    if not validTemp(degrees):
-        return "invalid temperature"
-    if degrees < 0:
-        if degrees < -273:
-            print("This temperature is likely Fahrenheit.")
-        else:
-            print("This temperature is either Celsius or Fahrenheit.")
-    else:
-        print("This temperature is in Kelvin, Celsius, or Fahrhenheit.")
+def _filter_even(number):
```
Comment (lyashevska) [inline code_modularization/00_title.md:334]: Good example, but again why not just using validTemp()?

**Review by lyashevska — COMMENTED** [review]
Very nice work, @DaniBodor. Thanks! I have only a couple of comments.

File: code_modularization/00_title.md (hunk 1 above)
Comment (DaniBodor) [reply code_modularization/00_title.md:313]: I didn't like the old example. The module that was extracted was basically just an if statement.
I couldn't think of another example that fell within the temperature conversion example set.

Comment (lyashevska) [reply code_modularization/00_title.md:313]: Thanks, I am happy now. Slides look great!

**Review by lyashevska — APPROVED** [review]
Thanks, it looks very clean.

**Pull request merged**